### PR TITLE
Remove quinn-proto rustls config newtypes

### PR DESCRIPTION
**Pull request by Ralith**
These were not successful at removing rustls from our public API, and made specifying custom cryptographic configurations unintuitive. Removing them reduces the scope of our public API while preserving the friendliness of the high-level quinn API.

### CODE REVIEW

File: quinn-proto/src/crypto.rs [hunk 1]
```diff
@@ -87,6 +87,11 @@ pub trait ClientConfig<S>
 where
     S: Session,
 {
+    /// Construct the default configuration
```
Comment (djc) [inline quinn-proto/src/crypto.rs]: Should we use `Default` for this, instead?

File: quinn-proto/src/crypto/rustls.rs [hunk 2]
```diff
@@ -238,170 +175,23 @@ impl crypto::ClientConfig<TlsSession> for ClientConfig {
...
-    }
-}
-
-impl DerefMut for ServerConfig {
-    fn deref_mut(&mut self) -> &mut Arc<rustls::ServerConfig> {
-        &mut self.0
-    }
-}
-
-/// A single TLS certificate
```
Comment (djc) [inline quinn-proto/src/crypto/rustls.rs]: Why did you move these into quinn?

File: quinn-proto/src/crypto.rs (hunk 1 above)
Comment (Ralith) [reply quinn-proto/src/crypto.rs]: We can't implement `Default` on `Arc<rustls::Client/ServerConfig>`.

File: quinn-proto/src/crypto/rustls.rs (hunk 2 above)
Comment (Ralith) [reply quinn-proto/src/crypto/rustls.rs]: With the newtypes removed from quinn-proto, they're only used in the quinn builders, so they seemed out of place here.

Comment (djc) [reply quinn-proto/src/crypto/rustls.rs]: I feel like they should live closer to the rustls implementation bits, so that they could also be swapped out along with them. Do you think that would make sense?

Comment (djc) [reply quinn-proto/src/crypto/rustls.rs]: They also seem like things that a quinn-proto user might still find useful if they wanted to bypass the tokio stuff.

Comment (Ralith) [reply quinn-proto/src/crypto/rustls.rs]: > I feel like they should live closer to the rustls implementation bits, so that they could also be swapped out along with them.

Other cryptographic protocols (e.g. noise) won't necessarily have analogous abstractions. For the specific case of other TLS implementations, we might prefer to expose a single abstract type that can be consumed by all backends, though that's pretty speculative.

> They also seem like things that a quinn-proto user might still find useful if they wanted to bypass the tokio stuff.

The sole function of these types was to allow certificates/keys to be constructed without leaving the crate and passed into the helper methods that were removed, and the analogous builder methods in quinn which have been retained. While there's nonzero value in providing those helpers to hypothetical quinn-proto users, I think the resulting quinn-proto API was gratuitously weird for both quinn and quinn-proto users who need to access the rustls configuration directly, and maintaining two separate incomplete abstractions over rustls configurations wasn't a great situation.

Meanwhile, the genuine rustls API is perfectly usable. The justification for wrapping it as all is to maximize approachability for a new high-level API user, but at the quinn-proto level I think the calculus is different.
